driver/k8s/logicalvolume_service.go

```diff
@@ -221,6 +245,75 @@ func (s *LogicalVolumeService) DeleteVolume(ctx context.Context, volumeID string
 	}
 }
 
+// CreateSnapshot creates a snapshot of existing volume.
+func (s *LogicalVolumeService) CreateSnapshot(ctx context.Context, sourceVol *topolvmv1.LogicalVolume, sourceVolID, sname, snapType, accessType string) (string, error) {
```

Comment:
keep it signature same as CreateVolume
```suggestion
func (s *LogicalVolumeService) CreateSnapshot(ctx context.Context, sourceVolID, sname, snapType, accessType string,  sourceVol *topolvmv1.LogicalVolume) (string, error) {

```